src/main/java/mekanism/common/util/MekanismUtils.java

```diff
@@ -777,8 +777,12 @@ public static void veinMineArea(IEnergyContainer energyContainer, Level world, B
                 continue;
             }
             BlockState targetState = world.getBlockState(foundPos);
-            FloatingLong destroyEnergy = destroyEnergyFunction.apply(targetState.getDestroySpeed(world, foundPos))
-                  .multiply(distanceMultiplier.applyAsDouble(WorldUtils.distanceBetween(pos, foundPos)));
+            float hardness = targetState.getDestroySpeed(world, foundPos);
+            if (targetState.isAir() || hardness == -1) {
+                continue;
+            }
+            FloatingLong destroyEnergy = destroyEnergyFunction.apply(hardness)
+                  .multiply(distanceMultiplier.apply(WorldUtils.distanceBetween(pos, foundPos), targetState));
```

Comment:
Thoughts on if it this would make more sense to instead have `ModuleVeinMiningUnit#findPositions` return something like `Map<BlockPos, BlockVeinData>` with `BlockVeinData` being a record of `BlockState, int distance` and then use that path distance instead of calculating the physical distance here? I am conflicted about which one makes more sense to use here.

Reply:
So that would actually solve the whole ugly iteration variable thing too, and I think path distance does make more sense here. Although just marking those variables volatile might have fixed the scope issues, didn't test it though. Instead of BlockState why don't we just do isOre since that is what the util method needs anyway and we'd have calculated it when finding the positions anyway (and it shouldn't change with the state like the hardness)

Reply:
Yeah just having an Object2IntMap would be fine, I just suggested using a record as I couldn't remember if the map return is used anywhere else in which case maybe it would have made passing things further along more difficult and require making a new map rather than just storing a bit of extra data in the value. Either is fine.